pkg/configprofiles/profiles.go

```diff
@@ -85,28 +63,7 @@ var staticProfiles = map[string]configProfile{
 // NOTE: DO NOT MODIFY TASKS HERE. Task execution is identified by the
 // task ID; already-run tasks will not re-run. Add tasks at the end of
 // each config profile. See enableReplication() for an example.
-var virtClusterInitTasks = []autoconfigpb.Task{
```

Comment:
I hate to ask, but is it safe to reduce the number of tasks in a job?  

Perhaps it doesn't mater if we make it all the way to changing how this works.

However, given that this is empty, mind doing a little more cleanup and getting rid of all the various array appending below in favor of 3 package-level vars for the 3 tasks we still have?

Reply:
I'm inclined to just delete the whole package in a follow-up change.

WDYT?

Reply:
Works for me.

Reply:
We're okay with leaning on the "preview" status of PCR in 23.2 to do this without a deprecation period, right? 

seems cleaner to just back out the config profiles now than keep them around deprecated.

Reply:
🤔 Since the configuration profiles are a one-time thing at startup, perhaps one thing we'll want to do is make sure that the configuration flags are still _accepted_.

Reply:
I donno, accepted but don't do anything seems worse to me? Saying "unknown flag" clues you in to the fact you need to read the release notes and see that your `--config=replication-source` cluster needs a settings flip if you want to talk to the app tenant instead of system.

Reply:
I'm thinking I'll close this PR and this can merge in #119366 since I'm not sure I like the intermediate state of having tasks with different steps much.